[Hub Generated] Publish private branch 'release-support-Microsoft.Support-2022-09-01-preview'

raghumunukutla opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2Fraghumunukutla%2Fazure-rest-api-specs-pr%2Frelease-support-Microsoft.Support-2022-09-01-preview...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/b321e1dc-5e07-4940-9cd6-a65ac6729a8b)

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

vimunuku committed:
First set of changes

vimunuku committed:
Changes for adding examples

vimunuku committed:
Minor change

vimunuku committed:
Changes as per comments

vimunuku committed:
Change to update swagger

vimunuku committed:
Changes for handling lintdiff errors

vimunuku committed:
Minor change

vimunuku committed:
Changes as per PR comments

vimunuku committed:
More changes

vimunuku committed:
More changes

raghumunukutla committed:
Adds base for updating Microsoft.Support from version stable/2020-04-01 to version 2022-09-01-preview

raghumunukutla committed:
Updates readme

raghumunukutla committed:
Updates API version in new specs and examples

raghumunukutla committed:
Changes for new preview api version

raghumunukutla committed:
Changes as per PR comments

raghumunukutla committed:
Minor change

vimunuku committed:
Changes for adding File attachment related paths, definitions and examples

vimunuku committed:
Changes as per review comments

vimunuku committed:
Changes as per PR comments

vimunuku committed:
More cahnges as per PR comments

vimunuku committed:
Changes as per PR comments from ARM team

vimunuku committed:
Changes for adding a new property as part of support ticket details model and updates for file properties

vimunuku committed:
Fixed duplicate operation Ids

vimunuku committed:
Fixing intendation

vimunuku committed:
Addressed all the errors by adding suppression rules and fixing the errors

vimunuku committed:
Changes for addressing PR comments

vimunuku committed:
Minor changes

vimunuku committed:
Addressed comments

vimunuku committed:
Minor change

vimunuku committed:
Fixed minor typo comments

vimunuku committed:
Minor fix

vimunuku committed:
Merge with latest of master

vimunuku committed:
Fixed prettier checks

vimunuku committed:
Changes as per checks

vimunuku committed:
Fixed PR comments

vimunuku committed:
Minor change and prettier check updates

vimunuku committed:
More cahges to address the errors from checks

vimunuku committed:
Changes as per checks

vimunuku committed:
Fixed few prettier checks

vimunuku committed:
More changes to fix model validation errors

vimunuku committed:
Minor fix

raghumunukutla committed:
Merge branch 'Azure:main' into release-support-Microsoft.Support-2022-09-01-preview

vimunuku committed:
Changes as per comments

vimunuku committed:
Adding a suppression for workspace getlist

raghumunukutla commented:
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/8647](https://github.com/Azure/azure-rest-api-specs-pr/pull/8647)

vimunuku committed:
Some model changes and associated updates to the examples

vimunuku committed:
Changes to fix errors from checks

vimunuku committed:
Changes to fix errors from checks

vimunuku committed:
Minor fixes in examples

raghumunukutla commented:
/pr RequestMerge

rkmanda commented:
You are at step 1 and its too early for u to request a merge. Pl request a merge once u get to step 4

raghumunukutla commented:
FYR, this PR was created based out of private preview PR with few new commits. Here is the private preview pr: https://github.com/Azure/azure-rest-api-specs-pr/pull/8647

rkmanda commented:
> FYR, this PR was created based out of private preview PR with few new commits. Here is the private preview pr: [Azure/azure-rest-api-specs-pr#8647](https://github.com/Azure/azure-rest-api-specs-pr/pull/8647)

Next time in adidtion to this also indicate which commits have the new changes so that the reviewer can focus only on those changes and the review will go faster

rkmanda commented on specification/support/resource-manager/readme.md:
```diff
@@ -26,18 +26,44 @@ These are the global settings for the Support API.
+  - code: RPC-Put-V1-11
+    from: Microsoft.Support/preview/2022-09-01-preview/support.json
+    reason: "Rule: Any Put MUST contain 200 and 201 return codes. Justification: We have similar implementation for previous versions, this would be a breaking change, hence suppressing it"
+  - code: RPC-Patch-V1-03
```
>RPC-Patch-V1-03 [](http://example.com/codeflow?start=10&length=15)

This breaks the patch semantics and must be implemented properly.

rkmanda replied:
Its not clear which property is being referred here. Can you provide more details and justification?

raghumunukutla replied:
Below are the properties that are referred as errors in private preview PR:

![image](https://github.com/Azure/azure-rest-api-specs/assets/55466801/f75eb399-f0fd-4180-926e-0e16c1e2aa67)


As part of this version, we are doing some minor model updates and added a new API fileworkspace. There were multiple meetings before coming up with design for fileworkspace and implemented accordingly and all other apis are same as previous versions. A detailed explanation was provided to ARM reviewer for the new additions and got an approval.

visingla-ms replied:
Sorry, I didn't understand this part: "These properties are not supported by our backend systems as per our business logic.". Do you mean, this PATCH API doesn't do anything today? Because if it actually updates status, contactDetails etc. properties in the resource today, the resource modeling needs to be correct for Patch request payload to match the same nested leveling with GET/PUT.

rkmanda commented on specification/support/resource-manager/readme.md:
```diff
@@ -26,18 +26,44 @@ These are the global settings for the Support API.
+  - code: RPC-Put-V1-11
+    from: Microsoft.Support/preview/2022-09-01-preview/support.json
+    reason: "Rule: Any Put MUST contain 200 and 201 return codes. Justification: We have similar implementation for previous versions, this would be a breaking change, hence suppressing it"
+  - code: RPC-Patch-V1-03
```
Having similar implementation in previous version is not a good rationale. Can u specify a valid reason for the design choice.

raghumunukutla replied:
As part of this version, we are doing some minor model updates and added a new API fileworkspace. There were multiple meetings before coming up with design for fileworkspace and implemented accordingly and all other apis are same as previous versions. A detailed explanation was provided to ARM reviewer for the new additions and got an approval for private preview pr.

raghumunukutla commented:
> > FYR, this PR was created based out of private preview PR with few new commits. Here is the private preview pr: [Azure/azure-rest-api-specs-pr#8647](https://github.com/Azure/azure-rest-api-specs-pr/pull/8647)
> 
> Next time in adidtion to this also indicate which commits have the new changes so that the reviewer can focus only on those changes and the review will go faster

Below are the new changes to the commit:

Adding secondaryConsentEnabled to the ProblemClassificationProperties

Adding ProblemScopingQuestions, secondaryConsent and supportPlanId to SupportTicketDetailsProperties

Adding secondaryConsent to UpdateSupportTicket

And below are the commit details:

https://github.com/Azure/azure-rest-api-specs/pull/25630/commits/94212c18448dee8d03cca56460d6020424b6037a

visingla-ms commented:
>         "responses": {

Put should have 200 response also, along with 201.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1544 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>         "responses": {

Same here: add 200 also in the response.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1623 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>         "responses": {

200?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1820 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>         "responses": {

200?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1922 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>             "description": "Successfully uploaded file content."

If Post doesn't suppose to return response body, then use 204 response, otherwise return a schema for 200.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1983 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>             "description": "Successfully uploaded file content."

If Post doesn't suppose to return response body, then use 204 response, otherwise return a schema for 200.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2036 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>           "type": "string"

Should it be enum?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2228 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>           "type": "string"

enum?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2780 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>           "properties": {

Why three levels of properties nesting?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2863 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>         "id": {

Why not refer to ProxyResource definition from common types: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json)?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2939 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>       "properties": {

Same here, use [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json) here from common types.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3000 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>           "type": "string"

string or number?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3043 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>           "description": "Size of the file to be uploaded",

Typo here?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3042 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented:
>           "type": "string"

string or number?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3057 in 06d8781. [](commit_id = 06d8781280e216e44d2d6bd28b7bb11d87945450, deletion_comment = False)

visingla-ms commented on specification/support/resource-manager/readme.md:
```diff
@@ -26,18 +26,44 @@ These are the global settings for the Support API.
+    from: Microsoft.Support/preview/2022-09-01-preview/support.json
+    reason: "Rule: The property {propertyName} in the request body either not apppear in the resource model or has the wrong level. Justification: We have similar implementation for previous versions, this would be a breaking change, hence suppressing it"
+  - code: RPC-Get-V1-05
+    from: Microsoft.Support/preview/2022-09-01-preview/support.json
```
If needed, apply the suppressions at particular definition, or code path level, not for whole file. Otherwise that will suppress all the future wrong usages as well.

raghumunukutla replied:
These suppressions are used for more than one definition and since we follow same pattern across, it made sense to have it for the whole file. So, keeping it as is.

visingla-ms replied:
Thank you. But for the below points, it can be considered:

1. RPC-Uri-V1-05: Why can't this be fixed for new properties being introduced, add pattern to new properties and apply linter suppressions only for existing properties if needed.

2. RPC-Patch-V1-03: For any new Patch APIs, this can be fixed, essentially should be fixed. So, apply suppression only for specific existing ones.

3. RPC-Get-V1-05: This also seems specific to FileWorkspaceDetails, so apply suppression only for this one.

Recommendation is to improve any new additions to the swagger and avoid previously non-recommended behavior in order to keep parity.

visingla-ms replied:
> Thank you. But for the below points, it can be considered:
> 
> 1. RPC-Uri-V1-05: Why can't this be fixed for new properties being introduced, add pattern to new properties and apply linter suppressions only for existing properties if needed.
> 2. RPC-Patch-V1-03: For any new Patch APIs, this can be fixed, essentially should be fixed. So, apply suppression only for specific existing ones.
> 3. RPC-Get-V1-05: This also seems specific to FileWorkspaceDetails, so apply suppression only for this one.
> 
> Recommendation is to improve any new additions to the swagger and avoid previously non-recommended behavior in order to keep parity.

Please respond to this, rest of the PR looks fine. Just these suppressions need to be defined correctly.

raghumunukutla replied:
Applied the changes to this suppressions specific to code path wherever applicable

raosuhas replied:
RPC-Get-V1-05 , Can you please explain the justification here ? Simply stating that we cannot support is not enough , what is the reason ?


For all the other ones , please suppress for specific lines as called out by vikas. The older implementations will not get flagged and we want to only suppress for specific instances that you are creating newly. 

If you have already fixed some of the suppressions then remove it from the readme file.

raosuhas replied:
please feel free to send me an email if you want to discuss this directly also.

raghumunukutla replied:
I have removed couple of suppression rules(RPC-Uri-V1-05 and RPC-Patch-V1-03), since we have taken care of those for new APIs. And for RPC-Get-V1-05, we explained in detail to the reviewer during private preview PR([[Hub Generated] Review request for Microsoft.Support to add version preview/2022-09-01-preview by raghumunukutla · Pull Request #8647 · Azure/azure-rest-api-specs-pr (github.com)](https://github.com/Azure/azure-rest-api-specs-pr/pull/8647)) and got a sign-off which is the reason why it was not mentioned in this PR.

I have emailed detailed explanation with screenshots from private preview PR and also the explanation.

vimunuku committed:
Changes for addressing comments

vimunuku committed:
Changes to address check failures

raghumunukutla commented:
> > ```
> >     "responses": {
> > ```
> 
> Put should have 200 response also, along with 201.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1544 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

As discussed offline, we are only adding 201 response based on our business requirements since replace is not supported.

raghumunukutla commented:
> > ```
> >     "responses": {
> > ```
> 
> Same here: add 200 also in the response.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1623 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

As discussed offline, we are only adding 201 response based on our business requirements since replace is not supported.

raghumunukutla commented:
> > ```
> >     "responses": {
> > ```
> 
> 200?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1820 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

As discussed offline, we are only adding 201 response based on our business requirements since replace is not supported.

raghumunukutla commented:
> > ```
> >     "responses": {
> > ```
> 
> 200?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1922 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

As discussed offline, we are only adding 201 response based on our business requirements since replace is not supported.

raghumunukutla commented:
> > ```
> >         "description": "Successfully uploaded file content."
> > ```
> 
> If Post doesn't suppose to return response body, then use 204 response, otherwise return a schema for 200.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:1983 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

Made changes to return 204

raghumunukutla commented:
> > ```
> >         "description": "Successfully uploaded file content."
> > ```
> 
> If Post doesn't suppose to return response body, then use 204 response, otherwise return a schema for 200.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2036 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

Made changes to return 204

raghumunukutla commented:
> > ```
> >       "type": "string"
> > ```
> 
> Should it be enum?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2228 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

This type is serviceType, for example "Databricks", "SQLServer". Since this is a big list and also new services keep adding, its not scalable to maintain enum, so we had it as string.

raghumunukutla commented:
> > ```
> >       "type": "string"
> > ```
> 
> enum?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2780 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

This type is serviceType, for example "Databricks", "SQLServer". Since this is a big list and also new services keep adding, its not scalable to maintain enum, so we had it as string.

raghumunukutla commented:
> > ```
> >       "properties": {
> > ```
> 
> Why three levels of properties nesting?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2863 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

Thanks for catching this, i updated swagger to have correct levels of definition.

raghumunukutla commented:
> > ```
> >     "id": {
> > ```
> 
> Why not refer to ProxyResource definition from common types: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json? Instead of defining id, name, type etc. here?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2939 in [5fe1f95](https://github.com/Azure/azure-rest-api-specs/commit/5fe1f9559ccf4c8adb2c9082441ae0397d893cac). [](commit_id = [5fe1f95](https://github.com/Azure/azure-rest-api-specs/commit/5fe1f9559ccf4c8adb2c9082441ae0397d893cac), deletion_comment = False)

Added the common types reference.

raghumunukutla commented:
> > ```
> >   "properties": {
> > ```
> 
> Same here, use https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json here from common types of ARM.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3000 in [5fe1f95](https://github.com/Azure/azure-rest-api-specs/commit/5fe1f9559ccf4c8adb2c9082441ae0397d893cac). [](commit_id = [5fe1f95](https://github.com/Azure/azure-rest-api-specs/commit/5fe1f9559ccf4c8adb2c9082441ae0397d893cac), deletion_comment = False)

Added the common types reference.

raghumunukutla commented:
> > ```
> >       "type": "string"
> > ```
> 
> string or number?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3043 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

Changed to number and updated examples as well.

raghumunukutla commented:
> > ```
> >       "description": "Size of the file to be uploaded",
> > ```
> 
> Typo here?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3042 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

Yes, I have fixed it.

raghumunukutla commented:
> > ```
> >       "type": "string"
> > ```
> 
> string or number?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3057 in [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450). [](commit_id = [06d8781](https://github.com/Azure/azure-rest-api-specs/commit/06d8781280e216e44d2d6bd28b7bb11d87945450), deletion_comment = False)

Changed to number and updated all examples.

vimunuku committed:
Changes to address multi level nesting comment

vimunuku committed:
Fixing prettier check and minor fix

visingla-ms commented:
>           "description": "Type of user consent.",

It doesn't convey that a service type is going in here.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2779 in 5f9ed4a. [](commit_id = 5f9ed4a6a3fd03abd50916f51258b974f6edf463, deletion_comment = False)

visingla-ms commented:
>           "type": "string"

From example, this looks like an enum: Yes/No.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2776 in 5f9ed4a. [](commit_id = 5f9ed4a6a3fd03abd50916f51258b974f6edf463, deletion_comment = False)

visingla-ms commented:
>           "type": "string"

It doesn't convey that a service type is going in here.

Also, it should at-least only allow users put a service type in standard form, which they can copy of Azure public docs.

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2228 in 5f9ed4a. [](commit_id = 5f9ed4a6a3fd03abd50916f51258b974f6edf463, deletion_comment = False)

visingla-ms commented:
>           "type": "string"

Yes/No enum?

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2224 in 5f9ed4a. [](commit_id = 5f9ed4a6a3fd03abd50916f51258b974f6edf463, deletion_comment = False)

visingla-ms commented:
>           "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/Resource"

Still it is "Resource", please refer to "ProxyResource" definition here (or TrackedResource as applicable).

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2952 in 5f9ed4a. [](commit_id = 5f9ed4a6a3fd03abd50916f51258b974f6edf463, deletion_comment = False)

visingla-ms commented:
>           "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/Resource"

Still it is "Resource", please refer to "ProxyResource" definition here (or TrackedResource as applicable).

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2870 in 5f9ed4a. [](commit_id = 5f9ed4a6a3fd03abd50916f51258b974f6edf463, deletion_comment = False)

visingla-ms commented:
>           "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/Resource"

Still it is "Resource", please refer to "ProxyResource" definition here (or TrackedResource as applicable).

---
Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3004 in 5f9ed4a. [](commit_id = 5f9ed4a6a3fd03abd50916f51258b974f6edf463, deletion_comment = False)

vimunuku committed:
Changes as per new comments

raghumunukutla commented:
> > ```
> >       "description": "Type of user consent.",
> > ```
> 
> It doesn't convey that a service type is going in here.
> 
> Also, it should at-least only allow users put a service type in standard form, which they can copy of Azure public docs.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2779 in [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463). [](commit_id = [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463), deletion_comment = False)

Updated the description, this value needs to be retrieved from the Problem Classification API response.

raghumunukutla commented:
> > ```
> >       "type": "string"
> > ```
> 
> From example, this looks like an enum: Yes/No.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2776 in [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463). [](commit_id = [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463), deletion_comment = False)

You are right, I updated this.

raghumunukutla commented:
> > ```
> >       "type": "string"
> > ```
> 
> It doesn't convey that a service type is going in here.
> 
> Also, it should at-least only allow users put a service type in standard form, which they can copy of Azure public docs.
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2228 in [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463). [](commit_id = [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463), deletion_comment = False)

Updated the description, this is something we are returning to customer.

raghumunukutla commented:
> > ```
> >       "type": "string"
> > ```
> 
> Yes/No enum?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2224 in [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463). [](commit_id = [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463), deletion_comment = False)

This is not Yes/No. As you can see from the example also, this value will be like "For faster resolution, allow Microsoft and Databricks to temporarily have read and write access to your Databricks workspace."

![image](https://github.com/Azure/azure-rest-api-specs/assets/55466801/6bbb6ed9-2f6c-44bd-93dd-71735d215edc)

raghumunukutla commented:
> > ```
> >       "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/Resource"
> > ```
> 
> Still it is "Resource", please refer to "ProxyResource" definition here (or TrackedResource as applicable).
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2952 in [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463). [](commit_id = [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463), deletion_comment = False)

Updated to proxyResource

raghumunukutla commented:
> > ```
> >       "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/Resource"
> > ```
> 
> Still it is "Resource", please refer to "ProxyResource" definition here (or TrackedResource as applicable).
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:2870 in [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463). [](commit_id = [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463), deletion_comment = False)

Updated to ProxyResource

raghumunukutla commented:
> > ```
> >       "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/Resource"
> > ```
> 
> Still it is "Resource", please refer to "ProxyResource" definition here (or TrackedResource as applicable).
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/preview/2022-09-01-preview/support.json:3004 in [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463). [](commit_id = [5f9ed4a](https://github.com/Azure/azure-rest-api-specs/commit/5f9ed4a6a3fd03abd50916f51258b974f6edf463), deletion_comment = False)

Updated to ProxyResource

visingla-ms commented:
Just check this comment: https://github.com/Azure/azure-rest-api-specs/pull/25630#discussion_r1336004018. Rest of the PR looks good. Next reviewer should be able to help you with reviewing this part.

vimunuku committed:
Changes to remove suppression at file level

vimunuku committed:
Changes for removing supression rules at file level

vimunuku committed:
Changes to apply suppression rule to the definition wherever applicable

vimunuku committed:
Removing couple of suppression rules as they are taken care for new api

raosuhas commented:
Approved suppression based on previous PR : https://github.com/Azure/azure-rest-api-specs-pr/pull/8647. New suppressions have been removed now

raych1 commented:
Approved SDK BC for Go and JS based on the previous approval of the PR in private spec repo which has same reported SDK BC.

vimunuku committed:
Changes to suppress TopLevelResourcesListBySubscription for fileworkspaces

vimunuku committed:
Reverting the suppression rule change

vimunuku committed:
Changing suppression code

raghumunukutla commented:
/pr RequestMerge

raosuhas commented:
Lint errors are failing incorrectly approving this
